docs/MPURegionsUsageGuide.md

```diff
@@ -0,0 +1,299 @@
+	CONFIG_ARMV8M_MPU_NREGIONS	8		Number of MPU regions	Chip Selection -> MPU Support -> Number of MPU Regions
+	------------------------------------------------------------------------------------------------------------------------------
+```
+
+## How to control MPU
+1. **How to add new MPU region**
+	- Logical mapping of MPU regions to their respective memory segments is done statically using below enumeration in common mpu.h header file (os/include/tinyara/mpu.h).
+	```c
+	enum MPU_REG_NUM {
+	#ifdef CONFIG_APP_BINARY_SEPARATION
+	#ifdef CONFIG_SUPPORT_COMMON_BINARY
+	#ifdef CONFIG_OPTIMIZE_APP_RELOAD_TIME
+		MPU_REG_NUM_COM_LIB_TXT,	/* Common Binary */
+		MPU_REG_NUM_COM_LIB_RO,
+		MPU_REG_NUM_COM_LIB_DATA,
+	#else
+		MPU_REG_NUM_COM_LIB,
+	#endif
+	#endif
+	#ifdef CONFIG_OPTIMIZE_APP_RELOAD_TIME
+		MPU_REG_NUM_APP_TXT,		/* Apps */
+		MPU_REG_NUM_APP_RO,
+		MPU_REG_NUM_APP_DATA,
+	#else
+		MPU_REG_NUM_APP,
+	#endif
+	#ifdef CONFIG_MPU_STACK_OVERFLOW_PROTECTION
+		MPU_REG_NUM_STK,		/* Stack */
+	#endif
+	#else
+		MPU_REG_NUM_UFLASH,
+		MPU_REG_NUM_URAM,
+	#endif
+		MPU_REG_NUM_MAX
+	};
+	```
+	- The ordering of MPU regions in this enum is kept as above taking into consideration below rule:
+		- For overlapping address range between MPU regions, attributes of higher MPU region number will override those set by lower numbered MPU region.
+	- Your new MPU region should be added the enum above.
+2. **How to register MPU region to real MPU hardware**
+	- Below list of functions is available in TizenRT for registering MPU region to MPU hardware.
+		- Several functions of mpu_xxxx format are available in architecture specific mpu.h file (eg. os/arch/arm/src/armv7-m/mpu.h) for one-time configuration of MPU regions.
+			- Example: For configuring MPU regions for User Flash and User Internal SRAM, below functions are available.
+			```c
+			/****************************************************************************
+			 * Name: mpu_user_flash
+			 *
+			 * Description:
+			 *   Configure a region for user program flash
+			 *
+			 * Params:
+			 *  region : MPU region number
+			 *  base   : MPU region base address
+			 *  size   : MPU region size
+			 ****************************************************************************/
+			static inline void mpu_userflash(uint32_t region, uintptr_t base, size_t size);
+			```
+			```c
+			/****************************************************************************
+			 * Name: mpu_userintsram
+			 *
+			 * Description:
+			 *   Configure a region as user internal SRAM
+			 *
+			 * Params:
+			 *  region : MPU region number
+			 *  base   : MPU region base address
+			 *  size   : MPU region size
+			 ****************************************************************************/
+			static inline void mpu_userintsram(uint32_t region, uintptr_t base, size_t size);
+			```
+		- For repeated configuration of a MPU region, a two-step process is required using below functions:
```

Comment:
Is there any use-case to use belows for user? If user uses below, is it ok to run our platform feature like app bin separation? @kishore-sn , could you confirm this?

Reply:
@sunghan-chang 
Yes. The two step process is necessary for app separation. First step will configure the values and store them in tcb. In second step the values are set in the mpu from tcb.

Reply:
@kishore-sn Yeah, I know it requires two steps to combine the values for mpu register and to set it at real hw.
But because of 2nd (to set real MPU hw), it looks wrong.
1. parameter
  That function's parameter is the tcb, that function takes the value from fixed location and the purpose of that location (array) is fixed. User can't ust that function for user.
2. naming
  The function name is `up_set_mpu_app_configuration`. That's only for app configuration. It does not have the name for common usages.

Reply:
I have added generic function for setting mpu registers:
`up_mpu_set_register`